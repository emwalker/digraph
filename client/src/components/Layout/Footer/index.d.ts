import React from 'react';
declare const Footer: () => React.JSX.Element;
export default Footer;
