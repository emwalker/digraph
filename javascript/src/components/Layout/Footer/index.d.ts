declare const Footer: () => JSX.Element;
export default Footer;
