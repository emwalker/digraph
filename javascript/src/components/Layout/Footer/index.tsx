import React from 'react'
import { Link } from 'found'

import styles from './styles.module.css'

const Footer = () => (
  <footer className={styles.footer}>
    <div className="container-lg clearfix my-3 px-3 px-md-6 px-lg-3 my-6 pt-2 border-top">
      <p className="mb-2">
        <Link
          className="gray-link"
          to="/terms-of-use"
          test-id="terms-of-use"
        >
          Terms
        </Link>
        ・
        Software available under the MIT
        {' '}
        <a
          className="gray-link"
          href="https://github.com/emwalker/digraph/blob/master/LICENSE.md"
        >
          license
        </a>
        . © Eric Walker.
      </p>
    </div>
  </footer>
)

export default Footer
