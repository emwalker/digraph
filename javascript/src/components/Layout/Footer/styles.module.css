.footer {
  position: absolute;
  width: 100%;
  bottom: 0;
}
