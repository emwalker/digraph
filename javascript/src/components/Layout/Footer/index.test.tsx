import React from 'react'
import { shallow } from 'enzyme'

import Footer from '.'

const props = {}

describe('<Footer />', () => {
  const wrapper = shallow(<Footer {...props} />)

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('has a link to the terms of use', () => {
    expect(wrapper.find('[test-id="terms-of-use"]').exists()).toBeTruthy()
  })
})
