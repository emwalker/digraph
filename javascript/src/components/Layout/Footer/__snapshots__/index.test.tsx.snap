// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders 1`] = `
<footer
  className="footer"
>
  <div
    className="container-lg clearfix my-3 px-3 px-md-6 px-lg-3 my-6 pt-2 border-top"
  >
    <p
      className="mb-2"
    >
      <Link
        className="gray-link"
        test-id="terms-of-use"
        to="/terms-of-use"
      >
        Terms
      </Link>
      ・ Software available under the MIT
       
      <a
        className="gray-link"
        href="https://github.com/emwalker/digraph/blob/master/LICENSE.md"
      >
        license
      </a>
      . © Eric Walker.
    </p>
  </div>
</footer>
`;
